### ISSUE TITLE
WorldTracker.lua Issues

### ISSUE THREAD

**Issue by the-m4a** [civfanatics/CQUI_Community-Edition#122]
I have only observed this with the BlackDeath scenario, however, the issue results in and endlessly looping reign of error (the FireTuner is quite the _entertaining_ sight with this one):

```
Runtime Error: A:\SteamLibrary\steamapps\common\Sid Meier's Civilization VI\DLC\Expansion2\UI\Replacements\WorldTracker_Expansion1.lua:65: function expected instead of nil
stack traceback:
	A:\SteamLibrary\steamapps\common\Sid Meier's Civilization VI\DLC\Expansion2\UI\Replacements\WorldTracker_Expansion1.lua:65: in function 'RealizeEmptyMessage'
	C:\Users\micha\OneDrive\Documents\My Games\Sid Meier's Civilization VI\Mods\CQUI_Community-Edition\Assets\UI\WorldTracker.lua:171: in function 'UpdateResearchPanel'
	A:\SteamLibrary\steamapps\common\Sid Meier's Civilization VI\DLC\Expansion2\UI\Replacements\WorldTracker_Expansion1.lua:26: in function 'UpdateResearchPanel'
	C:\Users\micha\OneDrive\Documents\My Games\Sid Meier's Civilization VI\Mods\CQUI_Community-Edition\Assets\UI\WorldTracker.lua:315: in function 'Refresh'
	C:\Users\micha\OneDrive\Documents\My Games\Sid Meier's Civilization VI\Mods\CQUI_Community-Edition\Assets\UI\WorldTracker.lua:362: in function 'FlushChanges'
	[C]: in function 'func'
	[C]: in function '(anonymous)'
```

This looks like it boils down to a few things:
- The CQUI Version of Worldtracker.xml has the EmptyPanel control commented out, and the CQUI version of worldtracker_expansion1.lua is _not_ loaded or taking priority over the basegame version of the same file.
- The CQUI version of WorldTracker.lua is missing some of the updates that have been made by Firaxis to their WorldTracker.lua.  Doing a file compare between the two revealed a lot of differences... 

For what it's worth, the specific line of code (in function RealizeEmptyMessage) is this:
`if( IsChatHidden() and IsCivicsHidden() and IsResearchHidden() and next(kCrisisData) == nil) then
`
None of IsChatHidden, IsCivicsHidden, IsResearchHidden are defined in the CQUI version of WorldTracker.lua.

These were added at some point to the base file (Civ6 Base\Assets\UI\WorldTracker.lua):
```
-- ===========================================================================
--	The following are a accessors for Expansions/MODs so they can obtain status
--	of the common panels but don't have access to toggling them.
-- ===========================================================================
function IsChatHidden()			return m_hideChat;		end
function IsResearchHidden()		return m_hideResearch;	end
function IsCivicsHidden()		return m_hideCivics;	end

```

**Comment by Infixo** [comment]
@the-m4a 
WorldTracker has a special extension for CivBattleRoyale. Before we close the issue I'd like to have an understaning that we are not (yet) going to make it compatible with CBR scenario.

**Comment by Infixo** [comment]
I reviewed the PR assiciated with that and these comments. I have one question first - why exactly is WorldTracker overwritten? What features does CQUI provide with it? I am looking at the code and I really can't see what it does. And I don't pay much attention to WT during a game.

**Comment by the-m4a** [comment]
The only thing CQUi did with it was to add a tooltip with info about the research/civic, and remove the empty panel control for reasons I did not identify (perhaps it was a previous feature that was removed?)

Cqui's changes are something we can likely live without, what do you think?

**Comment by Infixo** [comment]
Exactly. That is why I asked what it did. I am tempted to simply remove it from CQUI.

**Comment by the-m4a** [comment]
I don't have a problem with that.  @alimulhuda @alexeyOnGitHub do you have opinions on this?

**Comment by alimulhuda** [comment]
I find the tooltip a decent QoL improvement. It allows seeing the improvements at a glance. Otherwise I would be mousing over each of the individual items to build together the picture of what is being unlocked, or I would have to go to the tech/civic tree view to see the tooltip summary.
I think it adds value and I do use it, but I'm not sure how much we'd be losing if we removed it. I would have a stronger opinion if I tried playing a game or two without it.

**Comment by the-m4a** [comment]
I need to look closer at what reenabling that empty panel will do.  If it does not cause issues then I think we wouldn't need the XML replacement and then the extension file adding the tool tips can stay

**Comment by Infixo** [comment]
I honestly don’t know what tooltip we are talking about. It allows seeing improvements? Where? Can someone post a screenshot pls?

**Comment by alimulhuda** [comment]
You get it when you mouse over the button for the tech/civic. It is the same tooltip you get when you are in the tech/civic tree view or choose tech/civic.

![image](https://user-images.githubusercontent.com/178540/91646915-80dd5d00-ea22-11ea-8d5c-b2213bdbe6b9.png)

**Comment by Infixo** [comment]
@alimulhuda Thank you. I got it now. Seems useful and also there is some heavy coding to enable it, so let's not waste it.

The EmptyPanel is used when all other panels are closed, it just displayes a message saying something like "here is the place for panels, actrivate them". No idea why CQUI removed it. It is harmless. So, after brining it back - there is really no need to replace .xml file. I removed in the PR.

Now there is only 1 extension file, tooltip and entire World Tracker seems to be working ok.
